### PR TITLE
TIP-764: redo mass edit

### PR DESCRIPTION
**Pull request by juliensnz**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

Full rework of the mass edit wizard for product and families.
This PR basically replace all php code related to mass edit to redo it in front end style.
The only task managed by the backend is now the conversion from grid filters to pqb filters and launching the job.

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ class EditCommonAttributes extends AbstractMassEditOperation
     protected $attributeLocale;
 
     /** @var string */
-    protected $attributeChannel;
+    protected $attributeChannel;z
```
Comment (StephanDollberg) [inline src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php]: @SamirBoulil  says to remove the `z`.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/common/select2/family.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    ],
+    function (
+        _,
+        FetcherRegistry,
+        UserContext,
+        i18n,
+        Routing
+    ) {
+        return {
+            getConfig: function (initialValue) {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/common/select2/family.js]: add comment

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/product/change-family.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        template,
+        initSelect2
+    ) {
+        return BaseOperation.extend({
+            template: _.template(template),
+            events: {
+                'change .family': 'updateModel'
+            },
+
+            reset: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/product/change-family.js]: comments

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/product/change-status.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        BaseOperation,
+        template
+    ) {
+        return BaseOperation.extend({
+            template: _.template(template),
+            events: {
+                'change .switch': 'updateModel'
+            },
+
+            reset: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/product/change-status.js]: comment

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/form.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            /**
+             * Get the operation module corresponding to the given parameter
+             *
+             * @param {string} operationCode
+             *
+             * @return {object}
+             */
+            getOperationExtension: function (operationCode) {
+                return _.find(this.extensions, (extension) => {
+                    return extension.options.config.label !== undefined && extension.getCode() === operationCode;
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/form.js]: remove label check

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/form.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+                            messenger.notify(
+                                'success',
+                                __(
+                                    this.config.launchedLabel,
+                                    {
+                                        operation: this.getOperationExtension(this.getCurrentOperation()).getLabel()
+                                    }
+                                )
+                            );
+                        })
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/mass-edit/form/form.js]: error case

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/read-only-form.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+/**
+ * Attribute extension in order to disable an attribute if the form is disabled
+ *
+ * @author Julien Sanchez <julien@akeneo.com>
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/read-only-form.js]: raaaaah

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/read-only-form.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        return BaseForm.extend({
+            template: _.template(attributeTemplate),
+            readOnly: false,
+
+            /**
+             * {@inheritdoc}
+             */
+            configure: function () {
+                this.listenTo(this.getRoot(), 'pim_enrich:form:field:extension:add', this.addFieldExtension);
+                this.listenTo(this.getRoot(), 'pim_enrich:form:update_read_only', function (readOnly) {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/read-only-form.js]: see what the actual fuck is going on

Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/read-only-form.js]: saw it

**Pull request merged**